datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxDrService.java

```diff
@@ -0,0 +1,273 @@
+                        stackV4Response.getCluster().getStatus());
+                return sdxDrFailed(sdxCluster, stackV4Response.getCluster().getStatusReason(), pollingMessage);
+            } else {
+                if (FINISHED.equals(flowState)) {
+                    LOGGER.info("Flow finished, but Backup/Restore is not complete: {}", sdxCluster.getClusterName());
+                    return sdxDrFailed(sdxCluster, "stack is in improper state", pollingMessage);
+                } else {
+                    return AttemptResults.justContinue();
+                }
+            }
+        }
+    }
+
+    private AttemptResult<StackV4Response> sdxDrFailed(SdxCluster sdxCluster, String statusReason, String pollingMessage) {
+        LOGGER.info("{} failed, statusReason: {}", pollingMessage, statusReason);
+        return AttemptResults.breakFor("SDX " + pollingMessage + " failed '" + sdxCluster.getClusterName() + "', " + statusReason);
+    }
+
+    private AttemptResult<StackV4Response> sdxDrSucceeded(StackV4Response stackV4Response) {
+        LOGGER.info("Database DR operation doe SDX cluster {} is successfull", stackV4Response.getCluster().getName());
+        return AttemptResults.finishWith(stackV4Response);
+    }
+
+    private boolean stackAndClusterAvailable(StackV4Response stackV4Response, ClusterV4Response cluster) {
+        return stackV4Response.getStatus().isAvailable()
+                && cluster != null
+                && cluster.getStatus() != null
+                && cluster.getStatus().isAvailable();
+    }
+
+    /**
+     * Updates the status of the database backup/restore operation.
+     * @param operationId Operation Id
+     * @param status Status of the operation
+     * @param failedReason Failure reason, if any.
+     */
+    public void updateDatabaseStatusEntry(String operationId, SdxDatabaseDrStatus.Status status, String failedReason) {
+        if (Strings.isNullOrEmpty(operationId)) {
+            return;
+        }
+        SdxDatabaseDrStatus drStatus = sdxDatabaseDrStatusRepository.findSdxDatabaseDrStatusByOperationId(operationId);
+        drStatus.setStatus(status);
+        if (!Strings.isNullOrEmpty(failedReason)) {
+            drStatus.setStatusReason(failedReason);
+        }
+        sdxDatabaseDrStatusRepository.save(drStatus);
+    }
+
+    /**
+     * Gets the status of the database backup operation.
+     * @param sdxCluster Sdx cluster on which the backup operation is performed.
+     * @param operationId Operation Id
+     * @return Backup status
+     */
+    public SdxDatabaseBackupStatusResponse getDatabaseBackupStatus(SdxCluster sdxCluster, String operationId) {
+        SdxDatabaseDrStatus drStatus = sdxDatabaseDrStatusRepository.findSdxDatabaseDrStatusByOperationId(operationId);
+        if ((drStatus == null) || (!drStatus.getSdxClusterId().equals(sdxCluster.getId()))
+                || (!drStatus.getOperationType().equals(SdxDatabaseDrStatus.SdxDatabaseDrStatusTypeEnum.BACKUP))) {
```

Comment:
this condition is a bit hard to read and also almost the same as in `getDatabaseRestoreStatus`
it would worth to refactor them into a method

Reply:
will do